torch/_inductor/utils.py

```diff
@@ -1727,6 +1727,27 @@ def is_wait(node):
     return type(node) == ir._WaitKernel
 
 
+def is_torchrec_collective(ir_node) -> bool:
+    # TODO: this is a temporary solution to ensure that we can identify torchrec's
+    # communication ops. But in order to allow better communication and computation
+    # overlap, torchrec's communication ops should be not used.
+    return getattr(ir_node, "python_kernel_name", None) in (
+        "torch.ops.torchrec.all_to_all_single.default",
+        "torch.ops.torchrec.reduce_scatter_tensor",
+        "torch.ops.torchrec.all_gather_into_tensor",
+    )
```

Comment:
I believe we can check without using string name, instead we can check `ir_node.op_overload in [torch.ops.torchrec.all_to_all_single.default, ...]`

Reply:
also does `torch.ops.torchrec.reduce_scatter_tensor` and `all_gather_into_tensor` also need `.default` at the end?

Reply:
Added `.default`. I tried using 
```
node.op_overload in (
    torch.ops.torchrec.all_to_all_single.default,
    torch.ops.torchrec.reduce_scatter_tensor.default,
    torch.ops.torchrec.all_gather_into_tensor.default,
)
``` 
but this is giving errors like below, so I kept using string names to check. Do you have other suggestions?
```
   File "/data/users/xuanzh/pytorch/torch/_inductor/utils.py", line 1736, in is_collective
     torch.ops.torchrec.all_to_all_single.default,
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   File "/data/users/xuanzh/pytorch/torch/_ops.py", line 1232, in __getattr__
     raise AttributeError(
 torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
 AttributeError: '_OpNamespace' 'torchrec' object has no attribute 'all_to_all_single'
```

Reply:
I think we can check something like `hasattr(torch.ops.torchrec, "all_to_all_single")`